[WIP] vim-tutor-mode changes

fmoralesc opened the pull request:
- Some assorted changes.
- Expect regions can be defined outside of the document, allowing for a simpler handling of mappings (should fix #4533, #6389, #4913, #7005).

TODO:
- [x] update syntax file.
- [x] update documentation (tutor/tutor.tutor)
- [x] disallow old mechanism for including 'expect' regions?

This supersedes #5066.

fmoralesc committed:
tutor: don't enable folds by default

Shougo committed:
vim-patch:7.4.2354

Problem:    The example that explains nested backreferences does not work
            properly with the new regexp engine. (Harm te Hennepe)
Solution:   Also save the end position when adding a state. (closes vim/vim#990)

https://github.com/vim/vim/commit/d563883a1fb5ec6cf4a2758c5e36ac1ff4e9bb3d

justinmk committed:
third-party: Bump libtermkey to 0.20

fmoralesc committed:
tutor: some tweaks to begginer's tutorial

fmoralesc committed:
tutor: disable old method for setting "expect" regions

tutor: remove movement mappings

fmoralesc committed:
tutor: allow metadata to exist outside of the documents.

this makes 'expect' regions simpler to handle.

brcolow commented on runtime/tutor/en/vim-01-beginner.tutor:
```diff
@@ -947,8 +947,10 @@ There are many resources online to learn more about vim. Here's a bunch of them:
+- *vim-galore*: https://github.com/mhinz/vim-galore
 
-If you prefer a book, *Practival Vim* by Drew Neil is recommended often.
+If you prefer a book, *Practival Vim* by Drew Neil is recommended often (and the forthcoming
```
Practival => Practical (typo). Also I am not sure if it is a forthcoming *version* or *sequel*, no big deal though.

fmoralesc committed:
tutor: update tutor-mode tutorial

fmoralesc committed:
tutor: update syntax

sampletext regions no longer supported

make sure tutorExpect is available

don't conceal code region delimiters

justinmk commented:
Still WIP?

fmoralesc commented:
@justinmk Almost done. There are a few visual glitches in the tutor/tutor.tutor file, and there might be something else. The main code is ready I think, though.

(Also, I think that once #5031 is merged, I'll want to use that instead of `matchaddpos()`, but that's for the TODO list.)

fmoralesc committed:
tutor: reorganize
